metals/src/main/scala/scala/meta/internal/metals/SavedFileSignatures.scala

```diff
@@ -0,0 +1,66 @@
+package scala.meta.internal.metals
+
+import java.io.IOException
+
+import scala.collection.concurrent.TrieMap
+
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.MD5
+import scala.meta.io.AbsolutePath
+
+class SavedFileSignatures {
+  private val previousCreateOrModify = TrieMap[AbsolutePath, String]()
```

Comment:
Could we reuse MutableMd5Fingerprints somehow? Maybe just save them in didSave with a flag  to MutableMd5Fingerprints and then we can get the last saved one instead of calculating again?

We can still have a separate collection in Compilations so that it's possible to clear it and force new compilations?

Reply:
For `didSave` we put them anyway there but for file watcher event like `CreateOrModify` we don't, and we don't want to since it stores content, which we don't need. Also there is no way to check that the caller of `compileFile/s` saved the fingerprint (there may be an old one there), so it becomes some hidden knowledge that you should.